src/main/java/org/truffleruby/core/encoding/EncodingManager.java

```diff
@@ -179,21 +176,7 @@ public synchronized DynamicObject replicateEncoding(Encoding encoding, String na
 
     @TruffleBoundary
     public static Charset charsetForEncoding(Encoding encoding) {
-        final String encodingName = encoding.toString();
-
-        if (encodingName.equals("ASCII-8BIT")) {
-            return StandardCharsets.ISO_8859_1;
-        }
-
-        if (encoding == ISO8859_16Encoding.INSTANCE) {
-            return ISO_8859_16.INSTANCE;
-        }
```

Comment:
Does `encoding.getCharset()` also handle ISO_8859_16 and ASCII-8BIT?

Reply:
`ASCII-8BIT` is handled. But I missed that we had a custom charset for `ISO_8859_16`, so I'll have to investigate that a bit more.

Reply:
`ISO_8859_16` is handled in jcodings as well. jcodings sets up a Charset SPI and registers an `ISO_8859_16` Charset. So, we can remove our local copy of that class. Thanks for pointing that out.